utils/cipher.c

```diff
@@ -46,7 +46,7 @@ status_t aes_decrypt(ta_cipher_ctx* cipher_ctx) {
   }
 
   // concatenate (Device_ID, timestamp)
-  snprintf((char*)nonce, IMSI_LEN + MAX_TIMESTAMP_LEN + 1, "%s-%ld", cipher_ctx->device_id, cipher_ctx->timestamp);
+  snprintf((char*)nonce, IMSI_LEN + MAX_TIMESTAMP_LEN + 1, "%s-%llu", cipher_ctx->device_id, cipher_ctx->timestamp);
```

Comment:
Can you make use of `PRIu64` defined in `<stdint.h>` for portability?

Reply:
Done.